src/index.js

```diff
@@ -247,7 +247,11 @@ function UseWalletProvider({
       const web3ReactConnector =
         connector &&
         connector.web3ReactConnector &&
-        connector.web3ReactConnector({ chainId, ...(connector.config || {}) })
+        connector.web3ReactConnector({
+          chainId,
+          providedEthereum,
```

Comment:
This function instantiates the `InjectedConnector`, and it's why we need to drill the provider down. Of course, with other connectors this will not be a problem as this parameter will not be destructured by the other functions.